File: scripts/algorithms/V/Valid Mountain Array/Valid Mountain Array.py
// Runtime: 153 ms (Top 95.7%) | Memory: 18.70 MB (Top 43.81%)

class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        if len(arr)<=2 or max(arr)==arr[0] or max(arr)==arr[len(arr)-1]:
            return False
        f=True
        for i in range(len(arr)-1):
            if f and arr[i]>=arr[i+1]:
                f=False
            if not f and arr[i]<=arr[i+1]:
                return False
        return True

File: scripts/algorithms/V/Valid Mountain Array/Valid Mountain Array.rs
// Runtime: 3 ms (Top 60.3%) | Memory: 2.15 MB (Top 67.9%)

impl Solution {
    pub fn valid_mountain_array(arr: Vec<i32>) -> bool {
        (1..arr.len())
            .find(|&i| arr[i - 1] >= arr[i])
            .map_or(false, |j| j > 1 && (j..arr.len()).all(|j| arr[j - 1] > arr[j]))
    }
}